08_education/digital_access.qmd

```diff
@@ -0,0 +1,939 @@
+      value = calculate_cov(.y),
+      envir = .GlobalEnv
+    )
+  )
+```
+
+We then calculate confidence intervals at the 95% level by adding/subtracting
+the MOE estimate from our digital access variables:
+
+```{r}
+#| label: calculate-confidence-intervals
+
+calculate_confidence_intervals <- function(data) {
+  data |>
+    mutate(across(
+      .cols = any_of(c(
+        "total_households_digital",
+        "total_people_in_households_digital",
+        "other_digital",
+        "black_digital",
+        "hispanic_digital",
+        "white_digital",
+        "income_less_than_50k_digital",
+        "income_more_than_50k_digital"
+      )),
+      .fns = list(
+        lb = ~ .x - get(paste0("moe_", cur_column())),
+        ub = ~ .x + get(paste0("moe_", cur_column()))
+      ),
+      .names = "{.col}_{.fn}"
+    ))
+}
+
+
+lst(
+  county_overall,
+  place_overall,
+  county_raceeth,
+  place_raceeth,
+  county_income,
+  place_income
+) %>%
+  map2(
+    .x = names(.),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = calculate_confidence_intervals(.y),
+      envir = .GlobalEnv
+    )
+  )
+```
+
+## Calculate the digital access metric
+
+```{r}
+#| label: calculate-digital-access-metric
+
+calculate_digital_access_metric <- function(data) {
+  data |>
+    mutate(across(
+      .cols = any_of(c(
+        "total_households",
+        "total_people_in_households",
+        "other",
+        "black",
+        "hispanic",
+        "white",
+        "income_less_than_50k",
+        "income_more_than_50k"
+      )),
+      .fns = list(
+        share_digital_access = ~ get(paste0(cur_column(), "_digital")) / .x,
+        share_digital_access_lb = ~ get(paste0(cur_column(), "_digital_lb")) / .x,
+        share_digital_access_ub = ~ get(paste0(cur_column(), "_digital_ub")) / .x
+      ),
+      .names = "{.fn}_{.col}"
+    )) |>
+    arrange(geoid, name, year) %>%
+    # replace instances with division by zero
+    mutate(across(everything(), ~ if_else(is.nan(.x) | is.infinite(.x), NA, .x)))
+}
+
+
+lst(
+  county_overall,
+  place_overall,
+  county_raceeth,
+  place_raceeth,
+  county_income,
+  place_income
+) %>%
+  map2(
+    .x = paste0(names(.), "_digital_access"),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = calculate_digital_access_metric(.y),
+      envir = .GlobalEnv
+    )
+  )
+
+map_dbl(county_overall_digital_access, ~ sum(is.nan(.x)))
+map_dbl(place_overall_digital_access, ~ sum(is.nan(.x)))
+map_dbl(county_raceeth_digital_access, ~ sum(is.nan(.x)))
+map_dbl(place_raceeth_digital_access, ~ sum(is.nan(.x)))
+map_dbl(county_income_digital_access, ~ sum(is.nan(.x)))
+map_dbl(place_income_digital_access, ~ sum(is.nan(.x)))
+```
+
+# Create data quality flags
+
+We only look at the CVs for the numerator, which is less precise than the
+denominator. We use 0.1, 0.2, and 0.5 as thresholds because the CV only accounts
+for sampling error in the numerator. This is stricter than for some other
+metrics.
```

Comment:
OK but the numbers themselves are determined based on distribution?

Reply:
This comment was actually outdated so I removed it. However, I updated the text to mention that we are using the same cutoffs for 1-year vs 5-year ACS